app/src/main/java/com/github/se/cyrcle/model/parking/ParkingRepository.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.se.cyrcle.model.parking
+
+interface ParkingRepository {
+  /**
+   * Get a new unique identifier for a parking
+   *
+   * @return a new unique identifier for a parking
+   */
+  fun getNewUid(): String
+
+  /**
+   * Initialize the repository
+   *
+   * @param onSuccess a callback that is called when the repository is initialized
+   */
+  fun init(onSuccess: () -> Unit = {})
+
+  /**
+   * Get all parkings
+   *
+   * @param onSuccess a callback that is called when the parkings are retrieved
+   * @param onFailure a callback that is called when an error occurs
+   */
+  fun getParkings(onSuccess: (List<Parking>) -> Unit = {}, onFailure: (Exception) -> Unit = {})
+
+  /**
+   * Get a parking by its identifier
+   *
+   * @param id the identifier of the parking
+   * @param onSuccess a callback that is called when the parking is retrieved
+   * @param onFailure a callback that is called when an error occurs
+   */
+  fun getParkingById(
+      id: String,
+      onSuccess: (Parking) -> Unit = {},
+      onFailure: (Exception) -> Unit = {}
+  )
+
+  /**
+   * Get parkings between two points
+   *
+   * @param start the start point (bottom left) of the area
+   * @param end the end point (top right corner) of the area
+   * @param onSuccess a callback that is called when the parkings are retrieved
+   * @param onFailure a callback that is called when an error occurs
+   */
+  fun getParkingsBetween(
+      start: Point,
+      end: Point,
+      onSuccess: (List<Parking>) -> Unit = {},
+      onFailure: (Exception) -> Unit = {}
+  )
+
+  fun getKClosestParkings(
```

Comment:
Maybe add documentation here: even if function name is quite clear, could be useful for future given that this is quite a specific function (for eg just quickly mention that k is the number of parkings we look for nearby)

Reply:
The documentation is inherited from the Interface but it's true that it doesn't include many details